feature_state.go

```diff
@@ -0,0 +1,163 @@
+package unleash
```

Comment:
Harvested API from Client. This is effectively a state that exposes methods that determine if a given feature/variant is enabled for that state. This isn't quite there yet in terms of architecture but it's closer to how this conceptually should work. Don't want to do more here because this PR is already quite large